core/src/test/java/com/scottlogic/datahelix/generator/core/walker/decisionbased/RowSpecTreeSolverTests.java

```diff
@@ -108,6 +116,138 @@ void createRowSpecs_whenRootNodeHasSomeDecisions_returnsRowSpecOfRoot() {
+        ConstraintNode root = TestConstraintNodeBuilder.constraintNode()
+            .where(fieldB).isInSet("1", "2")
+            .withDecision(
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("1", "2", "3"),
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("1", "2", "4"))
+            .build();
+        DecisionTree tree = new DecisionTree(root, fields);
+
+        //Act
+        Set<WeightedElement<RowSpec>> rowSpecs = rowSpecTreeSolver.createRowSpecs(tree).collect(Collectors.toSet());
+
+        //Assert
+        Set<WeightedElement<RowSpec>> expectedRowSpecs = new HashSet<>();
+        Map<Field, FieldSpec> option0 = new HashMap<>();
+        option0.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option0.put(fieldB, FieldSpecFactory.fromLegalValuesList(Arrays.asList("1", "2")));
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option0, Collections.emptyList()), 1));
+        Map<Field, FieldSpec> option1 = new HashMap<>();
+        option1.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option1.put(fieldB, FieldSpecFactory.fromLegalValuesList(Arrays.asList("1", "2")));
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option1, Collections.emptyList()), 1));
+
+        assertThat(
+            rowSpecs,
+            sameBeanAs(expectedRowSpecs));
+    }
+
+    @Test
+    void createRowSpecs_whenRootNodeHasNoWeightedDecisions_returnsCorrectlyWeightedRowSpecs() {
+        //Arrange
+        ConstraintNode root = TestConstraintNodeBuilder.constraintNode()
+            .where(fieldB).isNotNull()
+            .withDecision(
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("1"),
+                TestConstraintNodeBuilder.constraintNode()
+                    .where(fieldB).isInSet("2"))
+            .build();
+        DecisionTree tree = new DecisionTree(root, fields);
+
+        //Act
+        Set<WeightedElement<RowSpec>> rowSpecs = rowSpecTreeSolver.createRowSpecs(tree).collect(Collectors.toSet());
+
+        //Assert
+        Set<WeightedElement<RowSpec>> expectedRowSpecs = new HashSet<>();
+        Map<Field, FieldSpec> option0 = new HashMap<>();
+        option0.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option0.put(fieldB, FieldSpecFactory.fromLegalValuesList(Collections.singletonList("1")).withNotNull());
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option0, Collections.emptyList()), 1));
+        Map<Field, FieldSpec> option1 = new HashMap<>();
+        option1.put(fieldA, FieldSpecFactory.fromType(fieldA.getType()));
+        option1.put(fieldB, FieldSpecFactory.fromLegalValuesList(Collections.singletonList("2")).withNotNull());
+        expectedRowSpecs.add(new WeightedElement<>(new RowSpec(fields, option1, Collections.emptyList()), 1));
```

Comment:
This block is repeated a lot.  Maybe create a createTwoElementFieldSpecMap(field1, field2, field2LegalValues) method that call and pass in fieldA, fieldB, COllections/SingletoList("2") in this particular example.

Reply:
Good point; i've created some helper methods to streamline to tests - i think